ui/ui-components/utils/makeLocal.ts

```diff
@@ -0,0 +1,7 @@
+/**
+ * Sometimes you *really* need to decouple a child object form a parent in a react useState method
+ * TODO: Why is this require to break the object chain to rules?
+ */
+export function makeLocal(obj: any) {
+  return JSON.parse(JSON.stringify(obj));
+}
```

Comment:
@seancdavis any idea why this is required?  If we are trying to copy an array of an object (eg `property.filters: []` into a local copy with `useState()` so we can manipulate it on the page, there doesn't seem to be a copy (`Array.from(property.filters)`) or spread (`[...property.filters]`) method that really copies the data into a new array.  This was the nuclear option that got things working 💣